app/controllers/posts_controller.rb

```diff
@@ -5,7 +5,8 @@ class PostsController < ApplicationController
     skip_before_action :verify_authenticity_token
 
   def index
-    @posts = Post.order('created_at DESC')
+    @post = Post.where(:User_id => current_user.id).notpublished  
```

Comment:
arrays will be returned - so change this to posts

Reply:
I have replaced @post with @posts